### ISSUE TITLE
+code seems to occasionally reset after retrieving

### ISSUE THREAD

**Issue by tiller-tolbus** [urbit/urbit#6262]
Hi, I'm currently running some servers hosting approx. 60 ships on the network. Frequently I will boot a ship for someone, retrieve an access code for them, and send them this access code in an email so they can use Urbit. 

Three times in the past several months I've generated an access code for a user and they emailed back saying they were unable to log in with it. Each time, I was able to go back into the dojo history for their ship and see that the access code I had sent them was not in error -- it was generated from their ship. However, running `+code` again generates a new code, and when I email it to them again, it begins working. 

I hesitated to make this issue as I have no earthly idea how one would go about reproducing this bug. However, it is certainly a bug -- I have ruled out any possibility of user error.

**Comment by Fang-** [comment]
Just as a sanity check, could you verify from dojo history (if that's still visible, otherwise, maybe up-arrow or ctrl-r search for "code"), that there was no instance of `|code %reset` being run? If they used webterm, this may or may not close their session before it gives them to new code. (Alternatively, `|pass [%j %step ~]`, but surely anyone running that will know what they're doing.)

**Comment by tiller-tolbus** [comment]
Dojo history is not available, but I can verify from the time the issue was discovered that the ships were young and the only dojo history was `+azimuth-block` and `+code`.

**Comment by Fang-** [comment]
Realizing now one other case where this may happen: when booting with a multi-keyfile.

In the common L2 flow, planet invites come with keys preconfigured, so they can be used right away, without having to wait for networking keys to get set & propagate. For obvious security reasons though, claiming an invite still sets new keys, but gives you a keyfile containing _both_ the old (from invite) and newly generated keys, so that you can boot the planet right away, but have it move on to the new keys automatically once the L2 update comes through.

Since the `+code` is derived from the _active_ networking keys, this would cause it to change as soon as the new keys take effect.

Perhaps, if jael knows multiple networking keys, we always want to derive the code from the latest, even if it's not known good/in use yet? But that seems confusing in its own way...

**Comment by tiller-tolbus** [comment]
In the meantime, would it be a valid workaround to wait until the roller configures the new keys before grabbing the boot key?

**Comment by Fang-** [comment]
Yes, if the above was indeed the case for what you experienced, waiting for 1) the L2 invite claim transaction to go through, and 2) for the ship in question to hear about that transaction, should ensure you only get the "final"/latest code.

**Comment by tiller-tolbus** [comment]
@vvisigoth in my opinion it's still objectively a bug -- but as far as I'm concerned we have a workflow that gets around it.
